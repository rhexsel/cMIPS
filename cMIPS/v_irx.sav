[*]
[*] GTKWave Analyzer v3.3.86 (w)1999-2017 BSI
[*] Mon Jun  3 18:24:48 2019
[*]
[dumpfile] "/home/roberto/downloads/cMIPS.vcd"
[dumpfile_mtime] "Mon Jun  3 18:19:28 2019"
[dumpfile_size] 70254592
[savefile] "/home/roberto/cMIPS/v_irx.sav"
[timestart] 36366800000
[size] 1447 990
[pos] -1 -1
*-26.114582 35400000000 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
[treeopen] u_core.
[treeopen] u_simple_uart.
[sst_width] 212
[signals_width] 217
[sst_expanded] 1
[sst_vpaned_height] 302
@28
clk
@22
u_core.pc[31:0]
@28
[color] 2
u_core.annul_twice
[color] 2
u_core.interrupt_taken
[color] 1
u_simple_uart.irq
@23
u_core.exccode[4:0]
@200
-    decode, reg fetch
@22
u_core.rf_instruction[31:0]
@200
-    exec
@22
u_core.alu_inp_a[31:0]
u_core.alu_inp_b[31:0]
u_core.result[31:0]
u_core.v_addr[31:0]
@200
-  COP0
@22
[color] 2
u_core.cause[31:0]
[color] 3
u_core.status[31:0]
[color] 2
u_core.epc[31:0]
@200
-    memory
@28
u_core.b_sel[3:0]
@22
d_addr[31:0]
u_core.data_inp[31:0]
u_core.data_out[31:0]
@28
cpu_d_aval
u_core.mm_wrmem
@200
-    write-back
@28
u_core.wb_wreg
@24
u_core.wb_a_c[4:0]
@22
u_core.wb_c[31:0]
@200
-  UART
@28
[color] 2
u_simple_uart.u_uart.s_stat
@22
[color] 2
u_simple_uart.u_uart.status[7:0]
[color] 3
u_simple_uart.u_uart.ctrl[7:0]
@28
[color] 7
u_simple_uart.u_uart.s_rx
@820
[color] 7
u_simple_uart.u_uart.received[7:0]
@22
[color] 7
u_simple_uart.u_uart.received[7:0]
@28
[color] 5
u_simple_uart.u_uart.s_tx
@22
[color] 5
u_simple_uart.u_uart.txreg[7:0]
@28
[color] 3
u_simple_uart.u_uart.s_intwr
[color] 3
u_simple_uart.u_uart.interr[1:0]
[color] 1
u_simple_uart.u_uart.irq_all
[color] 5
u_simple_uart.u_uart.rx_done
[color] 5
u_simple_uart.u_uart.interr_rx_full
[color] 7
u_simple_uart.u_uart.interr_tx_empty
@200
-  reception
@28
[color] 2
u_simple_uart.u_uart.rxclk
[color] 7
u_simple_uart.u_uart.rxdat
[color] 5
u_simple_uart.u_uart.txdat
[color] 4
u_simple_uart.u_uart.txclk
@200
-  REMOTE (fake) UART
@820
[color] 5
u_uart_remota.recv[7:0]
[pattern_trace] 1
[pattern_trace] 0
