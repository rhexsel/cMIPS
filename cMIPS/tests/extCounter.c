// Testing the external counter is difficult because it counts clock cycles
// rather than instructions -- if the io/instruction latencies change then
// the simulation output also changes and comparisons become impossible.

#include "cMIPS.h"


// convert small integer (i<16) to hexadecimal digit
#define i2c(a) ( ((a) < 10)   ? ((a)+'0') : (((a)+'a')-10) )


#define N  6                    // must be less than 29
#define CNT_VALUE 0x40000040    // set count to 64 cycles

void main(void) {
  int i, increased, new, old, newValue;

  newValue  = CNT_VALUE;
  increased = TRUE;

  for (i=1; i < (N+1); i++) {   // repeat N rounds
    print(i);                   // print number of round
    // to_stdout( i2c(i) );     // print number of round
    // to_stdout('\n');

    newValue = CNT_VALUE + (i<<3);
    startCounter(newValue, 0);  // num cycles increases with i, no interrupts

    old = 0;

    do {

      if ( (new = readCounter()) > old) {
	increased = increased & TRUE;
	old = new;
	// print(new);          // print current count, not for automated tests
      } else {
	increased = FALSE;
      }

    } while ( (readCounter() & 0x3fffffff) < (newValue & 0x3fffffff) );
    // are we done yet?

    if (increased) {
      to_stdout('o');
      to_stdout('k');
    } else {
      to_stdout('e');
      to_stdout('r');
      to_stdout('r');
    }

    to_stdout('\n');
  }

}
