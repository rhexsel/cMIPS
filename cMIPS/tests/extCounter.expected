00000001
ok
00000002
ok
00000003
ok
00000004
ok
00000005
ok
00000006
ok
